packages/example-log-extension/src/providers/log-level.provider.ts

```diff
@@ -10,6 +10,8 @@ export class LogLevelProvider implements Provider<number> {
   constructor() {}
 
   value(): number {
+    const level = Number(process.env.LOG_LEVEL);
```

Comment:
-1000

Internal implementation should not access `process.env` directly, because it introduces hidden coupling that's difficult to discover (learn about its existence) and make it difficult to customize the behavior in specific places only, without affecting everybody else living in the same Node.js process.

If we want to access `process.env.LOG_LEVEL`, the such code should live somewhere in the main application file where we are constructing and booting an `Application` instance.

Reply:
To illustrate the `provider` concept for logging level, we need to have a `variable`. Let me see what else we can use instead of env var.

Reply:
The provider returns a value that is bound via the component. The user can override the log-level but re-binding the value to a different value explicitly in the app.